util/timeutil/time.go

```diff
@@ -259,3 +260,17 @@ func ParseTimeZone(s string) (*time.Location, error) {
 
 	return nil, ErrUnknownTimeZone.GenWithStackByArgs(s)
 }
+
+// Sleep is a function like time.Sleep(),
+// the difference is it will return early when ctx finished.
+func Sleep(ctx context.Context, d time.Duration) error {
```

Comment:
For now all funcs in `time.go` are related to timezones, so it's a bit weird to add `Sleep` here, how about:

* Either we could rename `time.go` to something like `tzutil.go` and add `Sleep` to a new `time.go`.
* Or we could add `Sleep` to `util/misc.go`

WDYT?

Reply:
I'm prefer the first one and think maybe named `time_zone.go` is better.

Updated.